src/encoded/types/cart.py

```diff
@@ -96,3 +108,47 @@ def get_or_create_cart_by_user(context, request):
         '@type': ['result'],
         '@graph': carts or [cart]
     }
+
+
+@view_config(context=Cart.Collection, request_method='PUT', permission='save-carts', name='put-cart')
+def create_cart_by_user(context, request):
+    userid = [
+        p.replace('userid.', '')
+        for p in request.effective_principals
+        if p.startswith('userid.')
+    ]
+    if not userid:
+        raise HTTPBadRequest()
+    else:
+        userid = userid[0]
+    user = request.registry[COLLECTIONS]['user'].get(userid)
+    if not user:
+        raise HTTPBadRequest()
+    cart_name = None
+    user_carts = get_cart_objects_by_user(request, userid)
+    visible_user_carts = filter_carts_to_visible(request, user_carts)
+    countable_carts = [
+        cart
+        for cart in visible_user_carts
+        if cart['status'] != 'disabled'
+    ]
+    cart_atids = [cart['@id'] for cart in visible_user_carts] if visible_user_carts else []
+    cart_status = request.json.get('status', None)
+    cart_name = request.json.get('name', '').strip()
+    # User writing a new cart; check for cart overflow and naming conflicts
+    if len(countable_carts) >= CART_USER_MAX:
+        msg = 'Users cannot have more than {} carts'.format(CART_USER_MAX)
+        raise HTTPBadRequest(explanation=msg)
+    elif next((cart for cart in visible_user_carts if (cart['status'] != 'deleted' and
```

Comment:
I think you can use `any` here, e.g.: 
```python
elif any(
    [
        cart
        for cart in visible_user_carts
        if cart_conditions
    ]
)

Reply:
Is there a big risk/benefit advantage to using `any` here?